blockchain/processing.py

```diff
@@ -606,9 +626,85 @@ def _execute_phase_4(self, config, phase_3_info):
 
             print "phase 4 executed"
 
-    def _execute_phase_5(self, config, phase_5_info):
+    def _execute_phase_5(self, config, verification):
         """ public, Bitcoin bridge phase """
-        print "phase 5 executed"
+        phase = 5
+        verification_record = verification['record']
+
+        verification_info = verification['verification_info']
+        verification_record['verification_info'] = verification_info
+
+        if validate_verification_record(verification_record, verification_info):
+            timestamp_db.insert_verification(verification_record)
+            verification_record['block_id'] = None
+            verification_record['origin_id'] = None
+            verification_db.insert_verification(verification_record)
+            print "phase 5 executed"
+
+    def _execute_timestamping(self, config):
+        hashes = []
+        verification_info = {
+            'verification_records': {},
+            'blockchain_type':"BTC"
+        }
+        pending_records = timestamp_db.get_pending_timestamp()
+
+        if not pending_records:
+            return
+
+        for r in pending_records:
+            hashes.append(r['signature']['hash'])
+            if r['origin_id'] not in verification_info['verification_records']:
+                verification_info['verification_records'][r['origin_id']] = {}
+            verification_info['verification_records'][r['origin_id']][r['timestamp_id']] = r['signature']['hash']
+
+
+        transaction_hash = final_hash(hashes,type=256)
+        verification_info['hash'] = transaction_hash
+
+        stamper = BitcoinTimestamper(self.service_config['bitcoin_network'], BitcoinFeeProvider())
+        bitcoin_tx_id = stamper.persist(transaction_hash)
+        bitcoin_tx_id = b2lx(bitcoin_tx_id).encode('utf-8')
```

Comment:
Yes that should be fixed once the new requirements document is installed with pip.